src/migration.py

```diff
@@ -84,7 +97,7 @@ def _create_document_meta(row: types_.TableRow) -> types_.DocumentMeta:
             "Internal error, no implementation for creating document meta with missing link in row."
         )
     return types_.DocumentMeta(
-        path=Path(*row.path[:-1]) / f"{row.path[-1]}.md", link=row.navlink.link, table_row=row
+        path=_generate_document_path(row), link=row.navlink.link, table_row=row
```

Comment:
Could it be `Path(*row.path).with_suffix(".md")`?

Reply:
Sorry I think the `with_suffix` will replace the content after the dot, so a `1. Introduction` will be converted to `1.md`. The current approach is probably safer.